### PR TITLE
Add support for external flash and sdram to imx rt series

### PR DESCRIPTION
**Pull request by MaureenHelm**
Adds support for the boot data, image vector table (IVT), and device configuration data (DCD) structures used by the imx boot ROM to boot an application from external xip flash and initialize the SDRAM. This increases the available flash and ram on the mimxrt1050_evk to 64 MB and 32 MB respectively.

### CODE REVIEW

**Review by jhqian — APPROVED** [review]
Tested boot from QSPI on rt1060 evk board and using SDRAM configured by DCD as well.

**Comment by hakehuang** [comment]
@MaureenHelm , from segger official release only hyper flash and qspi flash are supported. Besides, there are some hard code change need to download to hyperflash

**Review by agansari — APPROVED** [review]
Changes look okay.

Suggestions:
1. Add memory increase to documentation, boards .yaml
2. Set external flash/ram as default?

**Comment by MaureenHelm** [comment]
> Suggestions:
> 
> 1. Add memory increase to documentation, boards .yaml
> 2. Set external flash/ram as default?

Yes I plan to do both of these, but wanted to give you and others a chance to try it out before I change any defaults. Can you please check if there are any caching issues with Ethernet buffers in SDRAM?

File: soc/arm/nxp_imx/rt/Kconfig.soc [hunk 1]
```diff
@@ -123,4 +123,41 @@ config IPG_DIV
 	int "IPG clock divider"
 	range 0 3
 
+menuconfig BOOT_HEADER
```
Comment (galak) [inline soc/arm/nxp_imx/rt/Kconfig.soc]: Should this be NXP_IMX_RT_BOOT_HEADER or something like that

**Review by galak — COMMENTED** [review]
Generally looks good, only question was about the Kconfig symbol names.

File: soc/arm/nxp_imx/rt/Kconfig.soc (hunk 1 above)
Comment (MaureenHelm) [reply soc/arm/nxp_imx/rt/Kconfig.soc]: Done

**Comment by MaureenHelm** [comment]
Updated to include rt1020 now that #12285 is merged

**Review by galak — APPROVED** [review]

**Pull request merged**